jax/_src/random.py

```diff
@@ -1724,3 +1724,85 @@ def ball(
   g = generalized_normal(k1, p, (*shape, d), dtype)
   e = exponential(k2, shape, dtype)
   return g / (((jnp.abs(g) ** p).sum(-1) + e) ** (1 / p))[..., None]
+
+def vonmises(
+  key: KeyArray,
+  x: float,
+  kappa: float,
+  shape: Shape = (),
+  dtype: DTypeLikeFloat = dtypes.float_
+) -> Array:
+  """Draw samples from a von Mises distribution.
+
+  Reference:
+    Best, D. J., & Fisher, N. I. (1979). Efficient Simulation of the
+    von Mises Distribution. Applied Statistics, 28(2), 152.
+    https://doi.org/10.2307/2346732
+
+  Args:
+    key: a PRNG key used as the random key.
+    x: mean of the distribution.
+    kappa: concentration of the distribution.
+    shape: shape of the output array.
+    dtype: dtype of the output array.
+
+  Returns:
+    theta: A random array of specified shape and dtype."""
+  # Run Checks
+  key, _ = _check_prng_key(key)
+  if not dtypes.issubdtype(dtype, np.floating):
+    raise ValueError(f"dtype argument to `t` must be a float"
+                     f"dtype, got {dtype}.")
+  dtype = dtypes.canonicalize_dtype(dtype)
+  shape = core.canonicalize_shape(shape)
+
+  # Return Distribution
+  if kappa <= 1e-6:
```

Comment:
This requires kappa to be a static parameter. If that's not your intent, you could use `lax.cond` instead.